### PR TITLE
tests: refactor config tests service name

### PR DESCRIPTION
**Pull request by david-luna**
This PR is refactoring all test related to the resolution of the service name and version. It also moves the `test/fixtures/pkg-zero-conf-*` folders used to test into `test/config/fixtures` folder to keep al related files in the same scope.

### Checklist

- [ ] Implement code
- [x] refactor tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [ ] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: test/config/config.test.js [hunk 1]
```diff
@@ -6,13 +6,17 @@
...
+var os = require('os');
+
 const test = require('tape');
 
 const { normalize } = require('../../lib/config/config');
 const { CONFIG_SCHEMA, getDefaultOptions } = require('../../lib/config/schema');
 
 const { runTestFixtures } = require('../_utils');
-const { reviver, replacer } = require('./_json-utils.js');
+const { reviver, replacer } = require('./json-utils.js');
```
Comment (david-luna) [inline test/config/config.test.js:19]: Note for reviewer: nit from the previous PR #3768

File: test/config/config.test.js [hunk 2]
```diff
@@ -448,7 +452,7 @@ const testFixtures = [
       t.deepEqual(
         resolvedConfig.globalLabels,
         pairs,
-        'globalLabels is parsed correctly from environment (array)',
+        'globalLabels is parsed correctly from start options (array)',
```
Comment (david-luna) [inline test/config/config.test.js:455]: Note for reviewer: nit from the previous PR #3768

File: test/config/config.test.js [hunk 3]
```diff
@@ -577,7 +581,7 @@ const testFixtures = [
       t.deepEqual(
         resolvedConfig.ignoreUrlStr,
         ['str1'],
-        'string items of ignoreUrl are added to the right config (ignoreUrlStr)',
+        'string items of ignoreUrls are added to the right config (ignoreUrlStr)',
```
Comment (david-luna) [inline test/config/config.test.js:584]: Note for reviewer: nit from the previous PR #3768

File: test/config/config.test.js [hunk 4]
```diff
@@ -596,6 +600,267 @@ const testFixtures = [
...
+        conf.serviceVersion,
+        '1.2.3',
+        'serviceVersion was inferred from package.json',
+      );
+    },
+  },
+  {
+    name: 'pkg-zero-conf-none - serviceName/serviceVersion resolved to unknown if no package.json present',
+    script: (function () {
+      const path = require('path');
```
Comment (david-luna) [inline test/config/config.test.js:812]: Note for reviewer: when this [iife](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) executes the requires at the top are not resolved yet. So we need to these ones

**Review by trentm — APPROVED** [review]
